proxy/src/main/java/com/wavefront/agent/ChannelStringHandler.java

```diff
@@ -46,15 +53,27 @@ public ChannelStringHandler(Decoder<String> decoder,
     this.pointHandler = pointhandler;
     this.preprocessor = preprocessor;
     this.metadataHandler = metadataHandler;
+
+    // new property rate
+    String logRawDataRateProperty = System.getProperty("wavefront.proxy.lograwdata");
+    logRawDataRate = logRawDataRateProperty != null && NumberUtils.isNumber(logRawDataRateProperty) ? Double.parseDouble(logRawDataRateProperty) : 0.0d;
+
+    // make sure the rate fits between 0.0d - 1.0d
+    if(logRawDataRate < 0.0d) {
+      logRawDataRate = 0.0d;
+    }
+    else if(logRawDataRate > 1.0d) {
+      logRawDataRate = 1.0d;
+    }
+    if(logRawDataRate > 0.0d) {
+      rawDataLogger.info("Raw data logging is enabled with " + (logRawDataRate * 100) + "% sampling");
```

Comment:
Suggestion: making this configurable also through rawDataLogger.isLoggable(Level.FINEST) would make it possible to enable and disable the logger on the fly without restarting the proxy

Reply:
Applied to check the isLoggable - make it possible to dynamically enable/disable the feature.